lib/motion/command/android_setup.rb

```diff
@@ -0,0 +1,167 @@
+        $stderr.puts("Installing NDK : version #{@ndk_version}")
+        ndk_url = File.join(DL_GOOGLE, 'ndk', "android-ndk-#{@ndk_version}-darwin-x86_64.bin")
+        ndk_bin = File.join(@tmp_directory, 'ndk.bin')
+        download(ndk_url, ndk_bin)
+        system("chmod +x #{ndk_bin}")
+        system("cd #{@tmp_directory} && #{ndk_bin}")
+        FileUtils.mv(File.join(@tmp_directory, "android-ndk-#{@ndk_version}"), @ndk_directory)
+      else
+        $stderr.puts("Installed NDK is up to date.")
+      end
+    end
+
+    def current_ndk_version
+      file = File.open(File.join(@ndk_directory, 'RELEASE.TXT'), "rb")
+      version = file.read.strip.split(' ')[0]
+      file.close
+      version
+    end
+
+    def setup_sdk
+      $stderr.puts("-- Setup SDK...")
+      if system("which #{android_executable} > /dev/null")
+        $stderr.puts("Found installed SDK, trying to update...")
+      else
+        $stderr.puts("Installing SDK : version #{@sdk_version}")
+        sdk_url = File.join(DL_GOOGLE, "android-sdk_r#{@sdk_version}-macosx.zip")
+        zipped_sdk = File.join(@tmp_directory, "sdk.zip")
+        download(sdk_url, zipped_sdk)
+        extracted_sdk = File.join(@tmp_directory, 'sdk')
+        system("/usr/bin/unzip -q -a #{zipped_sdk} -d #{extracted_sdk}")
+        FileUtils.mv(File.join(extracted_sdk, 'android-sdk-macosx'), @sdk_directory)
+      end
+      
+      system("#{android_executable} update sdk --no-ui --filter #{PACKAGES_LIST.join(',')}")
+    end
+
+    def create_directory_structure
+      $stderr.puts("-- Create directory structure...")
+      FileUtils.mkdir_p(@tmp_directory)
+      $stderr.puts(@directory)
+      $stderr.puts(@tmp_directory)
+    end
+
+    def check_java
+      $stderr.puts("-- Checking Java is installed...")
+      unless system("which /usr/bin/javac > /dev/null")
+        die("[error] Couldn't find Java, please make sure you have java installed, we recommend java 1.6")
+      end
+    end
+
+    def check_env_variables
+      $stderr.puts("-- Checking ENV variables...")
+      if ENV['RUBYMOTION_ANDROID_SDK'] != @sdk_directory
+        $stderr.puts("[error] RUBYMOTION_ANDROID_SDK is incorrect, should be #{@sdk_directory}")
+        $stderr.puts("add `export RUBYMOTION_ANDROID_SDK=#{@sdk_directory}` to your ~/.profile")
+      end
+      if ENV['RUBYMOTION_ANDROID_NDK'] != @ndk_directory
+        $stderr.puts("[error] RUBYMOTION_ANDROID_NDK is incorrect, should be #{@ndk_directory}")
+        $stderr.puts("add `export RUBYMOTION_ANDROID_NDK=#{@ndk_directory}` to your ~/.profile")
+      end
+    end
+
+    def curl(cmd)
+      resp = `/usr/bin/curl --connect-timeout 60 #{cmd}`
+      if $?.exitstatus != 0
+        die("Error when connecting to the server. Check your Internet connection and try again.")
+      end
+      resp
+    end
+
+    def download(url, dest)
+      if system("which axel > /dev/null")
+        unless system("axel -n 10 -a -o '#{dest}' '#{url}'")
```

Comment:
Here we should be using the path to axel in the $PATH.

Probably something like this:

```
axel_path = `/usr/bin/which axel`
if $?.success?
  unless system("#{axel_path.strip} -n ...")
else
  curl(...)
end
```